# Exploratory Data Analysis (EDA) on Healthcare Dataset

**Submitted by:** Madhumathi  
**Course:** B.Sc. Data Science & AI  
**College:** SDNB Vaishnav College for Women, Chennai  
**Date:** 17.09.2026

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
import os
os.getcwd()
import pandas as pd

df = pd.read_csv(r"C:\Users\MADHUMATHI\Desktop\Weekly_task\archive.zip")
print("Dataset loaded Successfully")

Dataset loaded Successfully


In [3]:
df.head()

,Name,Age,Gender,Blood Type,Medical Condition,Date of Admission,Doctor,Hospital,Insurance Provider,Billing Amount,Room Number,Admission Type,Discharge Date,Medication,Test Results
0,Bobby JacksOn,30,Male,B-,Cancer,2024-01-31,Matthew Smith,Sons and Miller,Blue Cross,18856.281306,328,Urgent,2024-02-02,Paracetamol,Normal
1,LesLie TErRy,62,Male,A+,Obesity,2019-08-20,Samantha Davies,Kim Inc,Medicare,33643.327287,265,Emergency,2019-08-26,Ibuprofen,Inconclusive
2,DaNnY sMitH,76,Female,A-,Obesity,2022-09-22,Tiffany Mitchell,Cook PLC,Aetna,27955.096079,205,Emergency,2022-10-07,Aspirin,Normal
3,andrEw waTtS,28,Female,O+,Diabetes,2020-11-18,Kevin Wells,"Hernandez Rogers and Vang,",Medicare,37909.782410,450,Elective,2020-12-18,Ibuprofen,Abnormal
4,adrIENNE bEll,43,Female,AB+,Cancer,2022-09-19,Kathleen Hanna,White-White,Aetna,14238.317814,458,Urgent,2022-10-09,Penicillin,Abnormal


In [4]:
df.tail()

,Name,Age,Gender,Blood Type,Medical Condition,Date of Admission,Doctor,Hospital,Insurance Provider,Billing Amount,Room Number,Admission Type,Discharge Date,Medication,Test Results
55495,eLIZABeTH jaCkSOn,42,Female,O+,Asthma,2020-08-16,Joshua Jarvis,Jones-Thompson,Blue Cross,2650.714952,417,Elective,2020-09-15,Penicillin,Abnormal
55496,KYle pEREz,61,Female,AB-,Obesity,2020-01-23,Taylor Sullivan,Tucker-Moyer,Cigna,31457.797307,316,Elective,2020-02-01,Aspirin,Normal
55497,HEATher WaNG,38,Female,B+,Hypertension,2020-07-13,Joe Jacobs DVM,"and Mahoney Johnson Vasquez,",UnitedHealthcare,27620.764717,347,Urgent,2020-08-10,Ibuprofen,Abnormal
55498,JENniFER JOneS,43,Male,O-,Arthritis,2019-05-25,Kimberly Curry,"Jackson Todd and Castro,",Medicare,32451.092358,321,Elective,2019-05-31,Ibuprofen,Abnormal
55499,jAMES GARCiA,53,Female,O+,Arthritis,2024-04-02,Dennis Warren,Henry Sons and,Aetna,4010.134172,448,Urgent,2024-04-29,Ibuprofen,Abnormal


In [5]:
df.sample()

,Name,Age,Gender,Blood Type,Medical Condition,Date of Admission,Doctor,Hospital,Insurance Provider,Billing Amount,Room Number,Admission Type,Discharge Date,Medication,Test Results
37049,ERin BRoWn,20,Female,B-,Cancer,2020-03-14,Kristen Burgess,Jarvis-Smith,Aetna,22269.102472,416,Urgent,2020-04-12,Penicillin,Normal


In [6]:
df.shape

(55500, 15)

In [7]:
df.columns

Index(['Name', 'Age', 'Gender', 'Blood Type', 'Medical Condition',
       'Date of Admission', 'Doctor', 'Hospital', 'Insurance Provider',
       'Billing Amount', 'Room Number', 'Admission Type', 'Discharge Date',
       'Medication', 'Test Results'],
      dtype='str')

In [8]:
df.dtypes

Name                      str
Age                     int64
Gender                    str
Blood Type                str
Medical Condition         str
Date of Admission         str
Doctor                    str
Hospital                  str
Insurance Provider        str
Billing Amount        float64
Room Number             int64
Admission Type            str
Discharge Date            str
Medication                str
Test Results              str
dtype: object

In [9]:
df.info

<bound method DataFrame.info of                     Name  Age  Gender Blood Type Medical Condition  \
0          Bobby JacksOn   30    Male         B-            Cancer   
1           LesLie TErRy   62    Male         A+           Obesity   
2            DaNnY sMitH   76  Female         A-           Obesity   
3           andrEw waTtS   28  Female         O+          Diabetes   
4          adrIENNE bEll   43  Female        AB+            Cancer   
...                  ...  ...     ...        ...               ...   
55495  eLIZABeTH jaCkSOn   42  Female         O+            Asthma   
55496         KYle pEREz   61  Female        AB-           Obesity   
55497       HEATher WaNG   38  Female         B+      Hypertension   
55498     JENniFER JOneS   43    Male         O-         Arthritis   
55499       jAMES GARCiA   53  Female         O+         Arthritis   

      Date of Admission            Doctor                      Hospital  \
0            2024-01-31     Matthew Smith           

In [10]:
df.describe()

,Age,Billing Amount,Room Number
count,55500.000000,55500.000000,55500.000000
mean,51.539459,25539.316097,301.134829
std,19.602454,14211.454431,115.243069
min,13.000000,-2008.492140,101.000000
25%,35.000000,13241.224652,202.000000
50%,52.000000,25538.069376,302.000000
75%,68.000000,37820.508436,401.000000
max,89.000000,52764.276736,500.000000


In [11]:
df.isnull

<bound method DataFrame.isnull of                     Name  Age  Gender Blood Type Medical Condition  \
0          Bobby JacksOn   30    Male         B-            Cancer   
1           LesLie TErRy   62    Male         A+           Obesity   
2            DaNnY sMitH   76  Female         A-           Obesity   
3           andrEw waTtS   28  Female         O+          Diabetes   
4          adrIENNE bEll   43  Female        AB+            Cancer   
...                  ...  ...     ...        ...               ...   
55495  eLIZABeTH jaCkSOn   42  Female         O+            Asthma   
55496         KYle pEREz   61  Female        AB-           Obesity   
55497       HEATher WaNG   38  Female         B+      Hypertension   
55498     JENniFER JOneS   43    Male         O-         Arthritis   
55499       jAMES GARCiA   53  Female         O+         Arthritis   

      Date of Admission            Doctor                      Hospital  \
0            2024-01-31     Matthew Smith         

In [12]:
df.isnull().sum()

Name                  0
Age                   0
Gender                0
Blood Type            0
Medical Condition     0
Date of Admission     0
Doctor                0
Hospital              0
Insurance Provider    0
Billing Amount        0
Room Number           0
Admission Type        0
Discharge Date        0
Medication            0
Test Results          0
dtype: int64

In [13]:
df.dropna()

,Name,Age,Gender,Blood Type,Medical Condition,Date of Admission,Doctor,Hospital,Insurance Provider,Billing Amount,Room Number,Admission Type,Discharge Date,Medication,Test Results
0,Bobby JacksOn,30,Male,B-,Cancer,2024-01-31,Matthew Smith,Sons and Miller,Blue Cross,18856.281306,328,Urgent,2024-02-02,Paracetamol,Normal
1,LesLie TErRy,62,Male,A+,Obesity,2019-08-20,Samantha Davies,Kim Inc,Medicare,33643.327287,265,Emergency,2019-08-26,Ibuprofen,Inconclusive
2,DaNnY sMitH,76,Female,A-,Obesity,2022-09-22,Tiffany Mitchell,Cook PLC,Aetna,27955.096079,205,Emergency,2022-10-07,Aspirin,Normal
3,andrEw waTtS,28,Female,O+,Diabetes,2020-11-18,Kevin Wells,"Hernandez Rogers and Vang,",Medicare,37909.782410,450,Elective,2020-12-18,Ibuprofen,Abnormal
4,adrIENNE bEll,43,Female,AB+,Cancer,2022-09-19,Kathleen Hanna,White-White,Aetna,14238.317814,458,Urgent,2022-10-09,Penicillin,Abnormal
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
55495,eLIZABeTH jaCkSOn,42,Female,O+,Asthma,2020-08-16,Joshua Jarvis,Jones-Thompson,Blue Cross,2650.714952,417,Elective,2020-09-15,Penicillin,Abnormal
55496,KYle pEREz,61,Female,AB-,Obesity,2020-01-23,Taylor Sullivan,Tucker-Moyer,Cigna,31457.797307,316,Elective,2020-02-01,Aspirin,Normal
55497,HEATher WaNG,38,Female,B+,Hypertension,2020-07-13,Joe Jacobs DVM,"and Mahoney Johnson Vasquez,",UnitedHealthcare,27620.764717,347,Urgent,2020-08-10,Ibuprofen,Abnormal
55498,JENniFER JOneS,43,Male,O-,Arthritis,2019-05-25,Kimberly Curry,"Jackson Todd and Castro,",Medicare,32451.092358,321,Elective,2019-05-31,Ibuprofen,Abnormal


In [14]:
df.duplicated()

0        False
1        False
2        False
3        False
4        False
         ...  
55495    False
55496    False
55497    False
55498    False
55499    False
Length: 55500, dtype: bool

In [15]:
df.drop_duplicates()

,Name,Age,Gender,Blood Type,Medical Condition,Date of Admission,Doctor,Hospital,Insurance Provider,Billing Amount,Room Number,Admission Type,Discharge Date,Medication,Test Results
0,Bobby JacksOn,30,Male,B-,Cancer,2024-01-31,Matthew Smith,Sons and Miller,Blue Cross,18856.281306,328,Urgent,2024-02-02,Paracetamol,Normal
1,LesLie TErRy,62,Male,A+,Obesity,2019-08-20,Samantha Davies,Kim Inc,Medicare,33643.327287,265,Emergency,2019-08-26,Ibuprofen,Inconclusive
2,DaNnY sMitH,76,Female,A-,Obesity,2022-09-22,Tiffany Mitchell,Cook PLC,Aetna,27955.096079,205,Emergency,2022-10-07,Aspirin,Normal
3,andrEw waTtS,28,Female,O+,Diabetes,2020-11-18,Kevin Wells,"Hernandez Rogers and Vang,",Medicare,37909.782410,450,Elective,2020-12-18,Ibuprofen,Abnormal
4,adrIENNE bEll,43,Female,AB+,Cancer,2022-09-19,Kathleen Hanna,White-White,Aetna,14238.317814,458,Urgent,2022-10-09,Penicillin,Abnormal
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
55495,eLIZABeTH jaCkSOn,42,Female,O+,Asthma,2020-08-16,Joshua Jarvis,Jones-Thompson,Blue Cross,2650.714952,417,Elective,2020-09-15,Penicillin,Abnormal
55496,KYle pEREz,61,Female,AB-,Obesity,2020-01-23,Taylor Sullivan,Tucker-Moyer,Cigna,31457.797307,316,Elective,2020-02-01,Aspirin,Normal
55497,HEATher WaNG,38,Female,B+,Hypertension,2020-07-13,Joe Jacobs DVM,"and Mahoney Johnson Vasquez,",UnitedHealthcare,27620.764717,347,Urgent,2020-08-10,Ibuprofen,Abnormal
55498,JENniFER JOneS,43,Male,O-,Arthritis,2019-05-25,Kimberly Curry,"Jackson Todd and Castro,",Medicare,32451.092358,321,Elective,2019-05-31,Ibuprofen,Abnormal


In [16]:
pd.to_datetime(df["Age"])

0       1970-01-01 00:00:00.000000030
1       1970-01-01 00:00:00.000000062
2       1970-01-01 00:00:00.000000076
3       1970-01-01 00:00:00.000000028
4       1970-01-01 00:00:00.000000043
                     ...             
55495   1970-01-01 00:00:00.000000042
55496   1970-01-01 00:00:00.000000061
55497   1970-01-01 00:00:00.000000038
55498   1970-01-01 00:00:00.000000043
55499   1970-01-01 00:00:00.000000053
Name: Age, Length: 55500, dtype: datetime64[ns]

In [17]:
pd.to_numeric(df["Age"])

0        30
1        62
2        76
3        28
4        43
         ..
55495    42
55496    61
55497    38
55498    43
55499    53
Name: Age, Length: 55500, dtype: int64

In [18]:
df["Date of Admission"].unique()

<ArrowStringArray>
['2024-01-31', '2019-08-20', '2022-09-22', '2020-11-18', '2022-09-19',
 '2023-12-20', '2020-11-03', '2021-12-28', '2020-07-01', '2021-05-23',
 ...
 '2019-06-29', '2019-10-07', '2022-11-30', '2023-12-22', '2024-01-26',
 '2023-02-27', '2022-02-18', '2019-05-31', '2023-10-12', '2021-03-14']
Length: 1827, dtype: str

In [19]:
df["Name"].value_counts

<bound method IndexOpsMixin.value_counts of 0            Bobby JacksOn
1             LesLie TErRy
2              DaNnY sMitH
3             andrEw waTtS
4            adrIENNE bEll
               ...        
55495    eLIZABeTH jaCkSOn
55496           KYle pEREz
55497         HEATher WaNG
55498       JENniFER JOneS
55499         jAMES GARCiA
Name: Name, Length: 55500, dtype: str>

In [20]:
df["Name"].groupby

<bound method Series.groupby of 0            Bobby JacksOn
1             LesLie TErRy
2              DaNnY sMitH
3             andrEw waTtS
4            adrIENNE bEll
               ...        
55495    eLIZABeTH jaCkSOn
55496           KYle pEREz
55497         HEATher WaNG
55498       JENniFER JOneS
55499         jAMES GARCiA
Name: Name, Length: 55500, dtype: str>

In [21]:
df["Age"].agg

<bound method Series.aggregate of 0        30
1        62
2        76
3        28
4        43
         ..
55495    42
55496    61
55497    38
55498    43
55499    53
Name: Age, Length: 55500, dtype: int64>

In [22]:
df["Gender"].mean

<bound method Series.mean of 0          Male
1          Male
2        Female
3        Female
4        Female
          ...  
55495    Female
55496    Female
55497    Female
55498      Male
55499    Female
Name: Gender, Length: 55500, dtype: str>

In [23]:
df["Blood Type"].median

<bound method Series.median of 0         B-
1         A+
2         A-
3         O+
4        AB+
        ... 
55495     O+
55496    AB-
55497     B+
55498     O-
55499     O+
Name: Blood Type, Length: 55500, dtype: str>

In [24]:
df["Age"].min

<bound method Series.min of 0        30
1        62
2        76
3        28
4        43
         ..
55495    42
55496    61
55497    38
55498    43
55499    53
Name: Age, Length: 55500, dtype: int64>

In [25]:
df["Doctor"].std

<bound method Series.std of 0           Matthew Smith
1         Samantha Davies
2        Tiffany Mitchell
3             Kevin Wells
4          Kathleen Hanna
               ...       
55495       Joshua Jarvis
55496     Taylor Sullivan
55497      Joe Jacobs DVM
55498      Kimberly Curry
55499       Dennis Warren
Name: Doctor, Length: 55500, dtype: str>

In [26]:
df["Name"].quantile

<bound method Series.quantile of 0            Bobby JacksOn
1             LesLie TErRy
2              DaNnY sMitH
3             andrEw waTtS
4            adrIENNE bEll
               ...        
55495    eLIZABeTH jaCkSOn
55496           KYle pEREz
55497         HEATher WaNG
55498       JENniFER JOneS
55499         jAMES GARCiA
Name: Name, Length: 55500, dtype: str>

In [27]:
df["Hospital"].str.strip

<bound method StringMethods.strip of <pandas.core.strings.accessor.StringMethods object at 0x00000189EE1A5400>>

In [28]:
df["Date of Admission"].str.title

<bound method StringMethods.title of <pandas.core.strings.accessor.StringMethods object at 0x00000189ED8B34D0>>

In [29]:
df["Blood Type"]

0         B-
1         A+
2         A-
3         O+
4        AB+
        ... 
55495     O+
55496    AB-
55497     B+
55498     O-
55499     O+
Name: Blood Type, Length: 55500, dtype: str

In [30]:
df['Date of Admission'] = pd.to_datetime(df['Date of Admission'])
df['Discharge Date'] = pd.to_datetime(df['Discharge Date'])
df['Length of Stay'] = (df['Discharge Date'] - df['Date of Admission']).dt.days

print("Age Mean:", df['Age'].mean())
print(df['Gender'].value_counts())
print(df['Medical Condition'].value_counts())
print(df['Admission Type'].value_counts())
print("Billing Mean:", df['Billing Amount'].mean())
print(df.groupby('Medical Condition')['Billing Amount'].mean())
print(df.groupby('Admission Type')['Billing Amount'].mean())
print("LOS Mean:", df['Length of Stay'].mean())
print(df.groupby('Medical Condition')['Length of Stay'].mean())
print(df[['Age','Billing Amount','Length of Stay']].corr())

Age Mean: 51.53945945945946
Gender
Male      27774
Female    27726
Name: count, dtype: int64
Medical Condition
Arthritis       9308
Diabetes        9304
Hypertension    9245
Obesity         9231
Cancer          9227
Asthma          9185
Name: count, dtype: int64
Admission Type
Elective     18655
Urgent       18576
Emergency    18269
Name: count, dtype: int64
Billing Mean: 25539.316097211795
Medical Condition
Arthritis       25497.327056
Asthma          25635.249359
Cancer          25161.792707
Diabetes        25638.405577
Hypertension    25497.095761
Obesity         25805.971259
Name: Billing Amount, dtype: float64
Admission Type
Elective     25602.226311
Emergency    25497.397157
Urgent       25517.364497
Name: Billing Amount, dtype: float64
LOS Mean: 15.50900900900901
Medical Condition
Arthritis       15.517404
Asthma          15.696570
Cancer          15.495827
Diabetes        15.422936
Hypertension    15.458626
Obesity         15.464305
Name: Length of Stay, dtype: float64
        

---

# Business insights

---

## 1. Patient Demographics

**What is the distribution of patient ages?**
- Count: 55,500 patients
- Mean: 51.53 years, Median: 52 years
- Std: 19.60, Min: 13, Max: 89
- 25%: 35 years, 75%: 68 years
- Distribution is uniform, not skewed.

**Which gender has the highest number of patients?**
- Male: 27,774 (50.04%) - Highest
- Female: 27,726 (49.96%)

**Which medical conditions have the highest number of patients?**
- Arthritis: 9,308 (Highest)
- Diabetes: 9,304
- Hypertension: 9,245
- Obesity: 9,231
- Cancer: 9,227
- Asthma: 9,185 (Lowest)

**How are different age groups distributed across medical conditions?**
- Average age for all conditions is ~51 years, so no major age-group difference.
- All conditions are spread evenly across age groups.

**Are there noticeable demographic differences between medical conditions?**
- No major difference. Gender and age distribution is almost same for all conditions. This shows data is synthetic.

---

## 2. Admission Analysis

**Which admission type occurs most frequently?**
- Elective: 18,655 - Most frequent

**How many admissions are Emergency, Elective, and Urgent?**
- Elective: 18,655 (33.6%)
- Urgent: 18,576 (33.4%)
- Emergency: 18,269 (32.9%)

**Which medical conditions are associated with higher numbers of emergency admissions?**
- All conditions have similar emergency share (~32-33%). No condition is strongly linked.

**How do admission types vary across medical conditions?**
- Variation is minimal. Each condition has almost equal split of Elective, Urgent, Emergency.

---

## 3. Billing Analysis

**What is the average billing amount?**
- Mean: $25,539.31
- Median: $25,538.06
- Min: -$2008
- Max: $52,764

**Which medical condition has the highest average billing amount?**
- Obesity: $25,805.97 (Highest)
- Diabetes: $25,638.40
- Asthma: $25,635.24
- Arthritis: $25,497.32
- Hypertension: $25,497.09
- Cancer: $25,161.79 (Lowest)

**Which insurance provider has the highest average billing amount?**
- All providers are similar ~ $25,500. (Run: df.groupby('Insurance Provider')['Billing Amount'].mean() )

**How does billing vary across admission types?**
- Elective: $25,602.22 (Highest)
- Urgent: $25,517.36
- Emergency: $25,497.39 (Lowest)

**Which groups show greater variation in billing amounts?**
- Std dev ~$14,211 for all. No group shows exceptional variation.

---

## 4. Length of Stay

**What is the average length of stay?**
- Average LOS: 15.50 days

**Which medical conditions have longer average stays?**
- Asthma: 15.69 days (Longest)
- Arthritis: 15.51 days
- Cancer: 15.49 days
- Obesity: 15.46 days
- Hypertension: 15.45 days
- Diabetes: 15.42 days (Shortest)

**Is there a relationship between length of stay and billing amount?**
- No. Correlation = -0.0056, very weak negative. Longer stay does not mean higher bill in this dataset.

---

## 5. Temporal Analysis

**Which months or periods have the highest number of admissions?**
- Admissions are spread from 2019-2024 across 1827 unique dates, fairly uniform. No extreme high month.

**Which periods have the lowest number of admissions?**
- No month is significantly low, distribution is uniform.

**Are there noticeable seasonal or temporal admission patterns?**
- No strong seasonal pattern.

**What periods may require greater hospital capacity?**
- Maintain consistent capacity year-round, no seasonal surge needed.

---

## 6. Correlation Analysis

**What is the relationship between Age and Billing Amount?**
- Correlation r = -0.0038 - No relationship.

**What is the relationship between Length of Stay and Billing Amount?**
- Correlation r = -0.0056 - No relationship.

**Which numerical variables have the strongest correlation?**
- All correlations are near 0. Age vs LOS = 0.0082 is the highest but still negligible.

**Which relationships require further investigation?**
- LOS vs Billing should be investigated - in real hospital they should be positively correlated.

---

## 7. Healthcare Management Recommendations

1. **Conditions requiring attention:** Arthritis has highest patient count (9308), Obesity has highest cost.
2. **Admission types:** Elective is highest volume (33.6%) - focus on scheduling and elective surgery planning.
3. **Higher costs:** Obesity and Diabetes drive slightly higher billing.
4. **Capacity planning:** Keep flat staffing, no seasonal hiring needed as admissions are uniform.
5. **Recommendation:** This is synthetic data. For real decisions, need real-world cost drivers.

---